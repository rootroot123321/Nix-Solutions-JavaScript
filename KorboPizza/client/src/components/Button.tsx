import * as React from 'react';
import classNames from 'classnames';

interface ButtonProps {
    onClick?: () => void;
    type?: "button" | "submit" | "reset" | undefined;
    className?: string;
    outline?: boolean;
    children?: React.ReactNode;
}

const Button = ({onClick, type, className, outline, children}: ButtonProps) => {
    return (
        <button
            onClick={onClick}
            type={type ?? undefined}
            className={classNames('button', className, {
                'button--outline': outline,
            })}>
            {children}
        </button>
    );
};

export default Button;
