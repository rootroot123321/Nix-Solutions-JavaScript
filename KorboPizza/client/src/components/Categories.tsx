import * as React from 'react';

interface CategoriesProps {
    activeCategory: (null|number);
    items: Array<string>;
    onClickCategory: (index: (null|number)) => void;
}

const Categories: React.FC<CategoriesProps> = React.memo(function Categories({activeCategory = null, items = [], onClickCategory}: CategoriesProps) {
    return (
        <div className="categories">
            <ul>
                <li
                    className={activeCategory === null ? 'active' : ''}
                    onClick={() => onClickCategory(null)}>
                    Все
                </li>
                {items &&
                items.map((name, index) => (
                    <li
                        className={activeCategory === index + 1 ? 'active' : ''}
                        onClick={() => onClickCategory(index + 1)}
                        key={`${name}_${index}`}>
                        {name}
                    </li>
                ))}
            </ul>
        </div>
    );
});

export default Categories;
