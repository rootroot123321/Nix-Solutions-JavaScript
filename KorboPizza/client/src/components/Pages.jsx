import React from 'react';
import {useDispatch, useSelector} from "react-redux";
import {setPage} from "../redux/actions/pizzas";

const Pages = () => {
    const dispatch = useDispatch();
    const {page:currentPage, totalCount, limit} = useSelector(({pizzas}) => pizzas);
    const pageCount = Math.ceil(totalCount / limit);
    const pages = [];
    for(let i = 0; i < pageCount; i++) {
        pages.push(i + 1);
    }
    return (
        <div className="pages">
            {pages.map(page =>
                <div
                    key={page}
                    className={`page ${page === currentPage ? 'current_page' : ''}`}
                    onClick={() => dispatch(setPage(page))}>{page}</div>
            )}
        </div>
    );
};

export default Pages;