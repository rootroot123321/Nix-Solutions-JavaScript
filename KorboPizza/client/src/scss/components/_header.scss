@import '../variables';

.header {
  border-bottom: 1px solid $gray-line;
  padding: 40px 0;

  .container {
    display: flex;
    align-items: center;
    justify-content: space-between;
    @media (max-width: 768px) {
      flex-direction: column;
    }
  }

  @media (max-width: 768px) {
    &__cart {
      margin-top: 30px;
    }
  }

  &__logo {
    display: flex;

    img {
      margin-right: 15px;
    }

    h1 {
      color: #181818;
      font-size: 24px;
      letter-spacing: 0.01em;
      text-transform: uppercase;
      font-weight: 800;
    }

    p {
      color: #7b7b7b;
    }
  }
  &__navbar {
    display: flex;
    align-items: center;
    button {
      border-radius: 10px;
      margin: 0 5px
    }
    select {
      width: 50px;
      height: 40px;
      font-size: 1.5em;
      background-color: $orange;
      border: none;
      border-radius: 10px;
      color: white;
    }
  }
}
