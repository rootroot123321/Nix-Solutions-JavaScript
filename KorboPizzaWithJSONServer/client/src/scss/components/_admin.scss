@import '../variables';

.admin {
  display: flex;
  flex-direction: column;
  margin: 5%;

  form {
    display: flex;
    flex-direction: column;
    align-items: center;
    margin: 3% 0;
    @mixin input-style {
      background-color: rgba(255, 155, 0, 0.5);
      width: 35%;
      height: 30px;
      padding: 2%;
      font-size: 1em;
      margin-bottom: 1.5%;
      border-radius: 10px;
    }
    .checkbox {
      display: flex;
      flex-direction: row;
    }
    input:not([type='file']) {
      @include input-style;
    }

    select {
      @include input-style;
      padding: initial;
    }
  }
}
