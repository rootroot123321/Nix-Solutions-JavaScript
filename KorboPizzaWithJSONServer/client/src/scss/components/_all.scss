@import './header';
@import './button';
@import './categories';
@import './sort';
@import './pizza-block';
@import './auth';
@import './admin';
