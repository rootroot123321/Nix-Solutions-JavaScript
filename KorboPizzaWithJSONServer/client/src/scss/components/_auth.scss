@import '../variables';

.authorization {
  border: 1px solid #ddd;
  background-color: rgba(255, 150, 0, 0.2);
  text-align: center;
  height: 120%;
  padding: 10%;

  form {
    display: flex;
    flex-direction: column;
    align-items: center;
    margin-top: 5%;

    input {
      background-color: rgba(255, 155, 0, 0.5);
      width: 50%;
      height: 50px;
      padding: 2%;
      font-size: 1.5em;
      margin: 1.5%;
      border-radius: 10px;
    }
    span {
      margin-top: 10px;
      a {
        color: dodgerblue;
        margin-left: 10px;
      }
    }
  }
}
