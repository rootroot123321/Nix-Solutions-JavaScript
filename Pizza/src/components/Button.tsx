import * as React from 'react';
import classNames from 'classnames';

interface ButtonProps {
    onClick?: () => void;
    className?: string;
    outline?: boolean;
    children?: React.ReactNode;
}

const Button = ({onClick, className, outline, children}: ButtonProps) => {
    return (
        <button
            onClick={onClick}
            className={classNames('button', className, {
                'button--outline': outline,
            })}>
            {children}
        </button>
    );
};

export default Button;
