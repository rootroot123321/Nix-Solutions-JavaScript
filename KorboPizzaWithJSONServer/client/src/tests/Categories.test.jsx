import React from "react";
import { render, unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";
import pretty from "pretty";
import { Categories } from "../components";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import store from "../redux/store";

let container = null;
const categoriesRender = (activeCategory, onSelectCategory, categories) => {
  return (
    <Provider store={store}>
      <BrowserRouter>
        <Categories
          activeCategory={activeCategory}
          onClickCategory={onSelectCategory}
          items={categories}
        />
      </BrowserRouter>
    </Provider>
  );
};
const fakeCategories = ["Острые", "Колбасные", "Грибные", "Сырные", "Закрытые"];
const onClickCategory = jest.fn();

beforeEach(() => {
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

it("renders the categories data (unit)", () => {
  act(() => {
    render(categoriesRender(0, onClickCategory, fakeCategories), container);
  });
  let activeCategory = container.querySelector(".active");
  expect(activeCategory.textContent).toBe("Острые");
  const categories = container.querySelector(".categories > ul").children;
  for (let i = 0; i < fakeCategories.length; i++) {
    expect(categories[i + 1].textContent).toBe(fakeCategories[i]);
  }
  act(() => {
    categories[3].dispatchEvent(new MouseEvent("click", { bubbles: true }));
  });
  expect(onClickCategory).toHaveBeenCalledTimes(1);
  act(() => {
    render(categoriesRender(2, onClickCategory, fakeCategories), container);
  });
  expect(container.querySelector(".active").textContent).toBe("Грибные");
});

it("renders the categories data (snapshot)", () => {
  act(() => {
    render(categoriesRender(4, onClickCategory, fakeCategories), container);
  });
  let activeCategory = container.querySelector(".active");
  expect(pretty(activeCategory.textContent)).toMatchInlineSnapshot(
    `"Закрытые"`
  );
  const categories = container.querySelector(".categories > ul").children;
  for (let i = 0; i < fakeCategories.length; i++) {
    expect(pretty(categories[i + 1].textContent)).toMatchSnapshot();
  }
  act(() => {
    categories[3].dispatchEvent(new MouseEvent("click", { bubbles: true }));
  });
  act(() => {
    render(categoriesRender(4, onClickCategory, fakeCategories), container);
  });
  expect(
    pretty(container.querySelector(".active").textContent)
  ).toMatchInlineSnapshot(`"Закрытые"`);
});
