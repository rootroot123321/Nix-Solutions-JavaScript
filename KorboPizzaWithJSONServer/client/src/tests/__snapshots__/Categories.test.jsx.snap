// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the categories data (snapshot) 2`] = `"Острые"`;

exports[`renders the categories data (snapshot) 3`] = `"Колбасные"`;

exports[`renders the categories data (snapshot) 4`] = `"Грибные"`;

exports[`renders the categories data (snapshot) 5`] = `"Сырные"`;

exports[`renders the categories data (snapshot) 6`] = `"Закрытые"`;
